### PR TITLE
Remove Entity and EntityTest

### LINKED ISSUE #193: Remove deprecated Entity
~~This should be done in DataModel 2.0~~

### PR DESCRIPTION
**Pull request by Benestar**
This removes the whole `Entity` class and makes it an alias of `EntityDocument`

Fixes #193

### CODE REVIEW

File: src/Entity/Item.php [hunk 1]
```diff
@@ -21,7 +23,7 @@
  * @licence GNU GPL v2+
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  */
-class Item extends Entity implements StatementListHolder {
+class Item implements Comparable, EntityDocument, FingerprintHolder, StatementListHolder {
```
Comment (JeroenDeDauw) [inline src/Entity/Item.php]: That is very... alphabetical? :) Think EntityDocument should be first

Comment (Benestar) [inline src/Entity/Item.php]: Indeed :) where should Comparable go then? To the end? Honestly this shouldn't matter a lot ;)

Comment (JeroenDeDauw) [inline src/Entity/Item.php]: Does not matter a lot yes. However putting the least important one before the by far most important one is kinda odd.

**Comment by JeroenDeDauw** [comment]
Somewhat surprised by you submitting this, given my comment on #520. As mentioned there, I think we should keep these deprecated methods as long as they are still used. Even if no code breaks with a change such as this, we will still end up with having lots of unresolved method calls. That also leaves us with no easy way of finding usages of deprecated methods, as there are existing unresolved method calls that have nothing to do with deprecation.

**Comment by thiemowmde** [comment]
Breaking changes, breaking changes everywhere

**Comment by Benestar** [comment]
@JeroenDeDauw wrote
> Somewhat surprised by you submitting this, given my comment on #520. As mentioned there, I think we should keep these deprecated methods as long as they are still used. Even if no code breaks with a change such as this, we will still end up with having lots of unresolved method calls. That also leaves us with no easy way of finding usages of deprecated methods, as there are existing unresolved method calls that have nothing to do with deprecation.

Sorry that I didn't answer your comment on the other pull request. As far as I can see both methods have only one usage left in Wikibase.git so we should be able to get rid of that quite easily. I wouldn't consider that a blocker tbh.

**Comment by JeroenDeDauw** [comment]
This PR is removing more than those two methods though, including those still used more than once...

**Comment by Benestar** [comment]
I will have a look again on the still remaining usages though since @addshore removed the serialization stuff I think all usages have gone.

**Comment by JeroenDeDauw** [comment]
Usage search including WB.git, WikibasQuality and WikibaseQualityConstraints and their dependencies (except DM itself)

* removeLabel: 1
* removeDescription: 1
* getAliases: 3
* getAllAliases: 3
* addAliases: 11
* removeAliases: 2
* getDescriptions: 5
* getLabels: 11
* getDescription: 6
* getLabel: 7
* setLabels: 2
* setDescriptions: 1
* setAllAliases: 0
* copy: 20
* getClaims: 4
* isEmpty: 9
* clear: 2
* setLabel & setDescription: bleh, hard to find out since derivative is undeprecated

**Comment by thiemowmde** [comment]
I think I used this argument before in an other pull request: The fact that this deprecated class does exist does not block us from implementing better solutions that do not need it. Just do not use it.

**Comment by thiemowmde** [comment]
I tried again to review this patch, but it does way to many things in my opinion. Please split or close this.

**Pull request closed without merging**